sdk/provisioning/Azure.Provisioning/src/ResourceOfT.cs

```diff
@@ -24,7 +24,20 @@ public abstract class Resource<T> : Resource
         /// <summary>
         /// Gets the properties of the resource.
         /// </summary>
-        public T Properties { get; }
+        public T Properties
+        {
+            get
```

Comment:
we need to explode in all public properties that aren't valid for existing scenarios.  Like AddOutput, AssignProperty etc.